instrumentation/netty/netty-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/AbstractChannelHandlerContextInstrumentation.java

```diff
@@ -46,15 +44,14 @@ public void transform(TypeTransformer transformer) {
     public static void onEnter(
         @Advice.This ChannelHandlerContext channelContext,
         @Advice.Argument(0) Throwable throwable) {
-      if (throwable != null) {
-        if (channelContext.channel().hasAttr(AttributeKeys.CLIENT_CONTEXT)) {
-          Attribute<Context> clientContextAttr =
-              channelContext.channel().attr(AttributeKeys.CLIENT_CONTEXT);
-          NettyHttpClientTracer.tracer().endExceptionally(clientContextAttr.get(), throwable);
-        } else {
-          NettyHttpServerTracer.tracer()
-              .onException(Java8BytecodeBridge.currentContext(), throwable);
-        }
+      if (channelContext.channel().hasAttr(AttributeKeys.CLIENT_CONTEXT)) {
+        Attribute<Context> clientContextAttr =
+            channelContext.channel().attr(AttributeKeys.CLIENT_CONTEXT);
+        NettyHttpClientTracer.tracer().endExceptionally(clientContextAttr.get(), throwable);
+      } else if (channelContext.channel().hasAttr(AttributeKeys.SERVER_CONTEXT)) {
```

Comment:
Netty40 has exactly the same logic, could you change that also.

Reply:
good call